src/config.rs

```diff
@@ -93,15 +99,21 @@ impl Config {
                     .and_then(Value::as_str)
                     .filter(|s| !s.is_empty())
                     .map(Into::into);
+
                 state.config.trace_macro_diagnostics =
                     response.pop_front().as_ref().and_then(Value::as_bool).unwrap_or_default();
+
                 state.config.enable_proc_macros =
                     response.pop_front().as_ref().and_then(Value::as_bool).unwrap_or(false);
 
+                state.config.enable_linter =
+                    response.pop_front().as_ref().and_then(Value::as_bool).unwrap_or(true);
+
                 debug!("reloaded configuration: {:#?}", state.config);
 
                 state.proc_macro_controller.on_config_change(&mut state.db, &state.config);
                 state.analysis_progress_controller.on_config_change(&state.config);
+                state.db.on_config_change(&state.config);
```

Comment:
```suggestion
                LinterController::on_config_change(&state.config);
```

So `db` does not contain linter-specific logic. It doesn't have to keep any state. WDYT?

Reply:
Totally agree, done